packages/core-typings/src/IMessage/IMessage.ts

```diff
@@ -249,6 +249,8 @@ export const isEditedMessage = (message: IMessage): message is IEditedMessage =>
 	'_id' in (message as IEditedMessage).editedBy &&
 	typeof (message as IEditedMessage).editedBy._id === 'string';
 
+export const isSystemMessage = (message: IMessage): message is ISystemMessage => 't' in message;
```

Comment:
I'm not sure if we could only consider the message having `t` as a system message. 🤔 

Maybe we should also check the type of the message (`MessageTypesValues`)... WDYT?

Reply:
I'm refactoring how we define message types to validate this